python/lsst/meas/base/forcedPhotCcd.py

```diff
@@ -147,6 +153,47 @@ class ForcedPhotCcdConfig(pipeBase.PipelineTaskConfig,
             "objects attached."
         ),
     )
+    refCatStorageClass = lsst.pex.config.ChoiceField(
+        dtype=str,
+        allowed={
+            "SourceCatalog": "Read an lsst.afw.table.SourceCatalog.",
+            "DataFrame": "Read a pandas.DataFrame.",
+            "ArrowAstropy": "Read an astropy.table.Table saved to Parquet.",
+        },
+        default="SourceCatalog",
+        doc=(
+            "The butler storage class for the refCat connection. "
+            "If set to something other than 'SourceCatalog', the "
+            "'inputSchema'  connection will be ignored."
+        )
+    )
+    refCatIdColumn = lsst.pex.config.Field(
+        dtype=str,
+        default="diaObjectId",
+        doc=(
+            "Name of the column that provides the object ID from the refCat connection. "
+            "measurement.copyColumns['id'] must be set to this value as well."
+            "Ignored if refCatStorageClass='SourceCatalog'."
```

Comment:
Seeing all these conditionally ignored config fields makes me sad and wish we could do something else like have an optional `ConfigField` that can be set to `None`.